### DISCUSSION TITLE
LYWSD03MMC with bigger battery

### DISCUSSION THREAD

**Discussion by DavidStacer** [pvvx/ATC_MiThermometer#396] [Show and tell]
I had some CR2450 cells. They are physically much bigger than the CR2032 and 2.6x larger capacity. With some case modification, they fit in the LYWSD03MMC. 

Take the two very small screws out  (Torx T5) and remove the inside plastic piece that holds the existing CR2032 cell. Then lay the new cell on over top the existing hole placing the bottom right where the two metal tabs touch the outside of the cell and draw a fine pencil line around the cell. Using a rotary tool with a small sanding drum just sand to the marked circle.  Go slow and test fit the cell,  you can always sand more off.  

Remove the black foam rubber from the snap on back and grind off the ribs that are under there. I also taped over some of the exposed PCB test points that might get touched with the new larger cell.  

Reassemble everything and put the new cell in place. The back should snap on. It will be interesting to see how long this cell lasts.

![ATD CR2450-2](https://github.com/pvvx/ATC_MiThermometer/assets/16613848/595e661b-a2d1-4043-b2bf-e0c89cc45ba7)
![ATC CR2450](https://github.com/pvvx/ATC_MiThermometer/assets/16613848/f85c1564-abf9-47d2-bbe6-c0169a7fa983)

**Comment by scottcopus** [comment]
Hi @DavidStacer. Cool! How goes your mods using CR2450?  Is it too soon to extrapolate how long this bigger cell will last compared the original CR2032?  How many have you modded like this so far?  Have you made any other kinds of mods to your LYWSD03MMC's--such as external batteries or using the GPIO pins for extra features?

**Reply by pvvx** [reply]
Some thermometer models, for which alternative firmware has long existed, use CR2450 and other CRxxx by default.
[MJWSD05MMC](https://pvvx.github.io/MJWSD05MMC/) has a similar SoC and LCD driver...

CR2450 variants are more stable in temperature and capacity.

**Reply by pvvx** [reply]
CGG1 - CR2430.
TH-05 - 2 x AAA
[....](https://pvvx.github.io/)

[AAA Energizer Ultimate Lithium](https://battest.ru/results/) > 1500 mAh for low current devices.
Noname CR2032 - 38 mAh

**Reply by DavidStacer** [reply]
So far I have only modified one of them for the CR2450 and it been in service for a month. It still shows 100%.  From the specifications the CR2450 is 2.6 times the capacity of the CR2032.

I have another mod where I had some extra CR123a cells and made up an adapter for that cell.  That sensor was outside for two years under a solar radiation shield.  That still shows 99% left.  That sensor suffers from some corrosion issues and I occasionally have to wiggle the connections... That one was down to -20C and still working.

Currently working on a 3D printed holder for the CR123a cell that will stick out of the back with a small support structure to hold it upright.  The mod is a tube that mounts to the existing back plate. This will have much less exposed to the outside weather and should solve the corrosion issue.

**Reply by pvvx** [reply]
> This will have much less exposed to the outside weather and should solve the corrosion issue.

But it does not solve the issue of curious birds.  Birds have already pulled out such thermometers several times and thrown them into a puddle. :)  
The life of the thermometer, with the [graphs presented on the main page](https://github.com/pvvx/ATC_MiThermometer#reading-measurements-from-flash), ended just like that. Failed to restore.
The LCD glass was broken. But there is a command to turn off the LCD for less consumption. Did not help.

---

MJWSD05MMC with battery included in the box:

![image](https://github.com/pvvx/ATC_MiThermometer/assets/12629515/3f1f9513-687a-42c6-b144-90f5b842af56)

Software Version was updated each time as it exit, plus tests before publication.

**Reply by DavidStacer** [reply]
No bird problems  - some photos might help.

![Outside_temp1 - Copy](https://github.com/pvvx/ATC_MiThermometer/assets/16613848/2db3d759-b5dc-4659-9860-ebd03d42c240)

![Outside_temp2 - Copy](https://github.com/pvvx/ATC_MiThermometer/assets/16613848/a9be29d1-bbeb-4fe4-952d-4084dadc4450)

**Reply by pvvx** [reply]
I now attach the thermometer with clamps. :)

There is a forest around and there are a lot of different birds...

**Comment by Yoshi-523** [comment]
` DavidStacer ` 
I know this this thread is a few years old now but I wanted to commend you on this genius modification/hack. I have been able to upgrade both for my LYWSD03MMC to run and fit tougher very well the larger CR2450 battery.

All it needed was some basic cheapo dremol type tool I had laying around from Aldi, the cheap compact drill set, a drill (ideally slow setting as well as fast). I also had a set of small diamond dust sand paper compatible with drilled. I don’t want people to feel intimidated to attempt a relatively simple job like this.

First make sure to unscrew and take the circuit board out and in a safe palace while you are drilling and cutting the plastic housing. If you skip this step up are likely to end drilling for damaging the circuit board.

I had an excessive amount of tools required to do this job as my 1st one. I took great care, used the many tools. The second one I ran out of time and patience and basically butchered the hole in place. The outside cover plate cover plate covered up all the imperfections. Just be aware that you want to expand the hole bit by bit for the larger CR2450 battery to sit in very snuggly.

PLEASE BE SURE TO STCIKY TAPE TO COVER, ELECRCALLY EXOPSPDED CIRCUIT BOARD, CONTACTS OR SOLDER JOINTS OR PCB CONTACT POINTS OR DECREIT SURFACE MOUNED SOLDERD INTEGRATED CIRCIUTS COMPONENTS. (IC’S SUCH AS TINY CUBED OR RECTAGNES, OFTREN RESISTORS, INDICTORS, TRANSITORS THAT OFTEN AVE METAL EXPLOSED AT THE TOP (IN THIS CALSE NEEDITNG TO BE COVERED WITH STICKY TAKE FOR INSULATIONS AGAINS SHORTING THE BATTERY TO IT.
THIS IS BEFPRE INSTALLING THE LARGER BATTER (CR2450) AS YOU DONT WANT A SHORT CURCUIT. LET ME BE CLEAR THAT BY CUTTING OUT OF THE SOME EXTRA PLASTIC SUPPORT FIT FOT THE LARGER DIAMETRE CR2450 BATTERY, TNE ORIGINAL SUPPORT PLASTIC FOR THE ORIGINAL SMALLER DIAMETER CR2032 BATTERY.

Best regards,
Yoshi

**Comment by ilgrank** [comment]
There's an even easier way: just use a CR2050, no mod required :)
[https://it.aliexpress.com/item/1005006897315279.html](https://it.aliexpress.com/item/1005006897315279.html)

@DavidStacer : can you share your CR123 model or photo? Very curious about it, thanks!

**Reply by DavidStacer** [reply]
> There's an even easier way: just use a CR2050, no mod required :) https://it.aliexpress.com/item/1005006897315279.html
> 
> @DavidStacer : can you share your CR123 model or photo? Very curious about it, thanks!
Sorry, must have missed the notification on your question. Better late than never?
Discussion topic 434, link below.
https://github.com/pvvx/ATC_MiThermometer/discussions/434

**Comment by chinswain** [comment]
I found this adaptor if anyone has a 3D printer: https://makerworld.com/en/models/959075-adapter-battery-cr2450-xiaomi-mi-2-thermometer#profileId-930398
